opac/webapp/main/views.py

```diff
@@ -1140,8 +1140,10 @@ def _handle_html():
 
         try:
             html, text_languages = render_html(article, qs_lang)
-        except (ValueError, NonRetryableError, RetryableError):
+        except (ValueError, NonRetryableError):
```

Comment:
Faz sentido que a aplicação continue retornando 404 em alguns casos? Penso que ambas as exceções `NonRetryableError` e `RetryableError` representam falhas na integração com componentes essenciais para o funcionamento do site (Minio e Kernel), portanto deveriam ser traduzidas para 500. 

Penso ainda que o erro 404 deveria ser retornado apenas após consulta por um identificador inexistente na base de dados do OPAC. Faz sentido?

Reply:
@gustavofonseca eu acredito que faz sim sentido continuar retornando 404 para os casos em que de fato o documento/arquivo não exista no kernel e/ou no minio. A consulta ao ponto de integração só  é feita depois que o próprio OPAC garante a existência do artigo em sua base mas isso não garante que o recurso existe no ponto de integração.

O [trecho de código](https://github.com/scieloorg/opac/blob/837757a235a15bb1aaf71beba6d6376b76c1a7e9/opac/webapp/main/views.py#L66-L68) tenta recuperar um recurso que pode não existir, isso pode acontecer por _n_ fatores que vão desde o arquivo removido no kernel/minio ou até mesmo uma URL registrada de forma errada na base do OPAC.

Reply:
@gustavofonseca @joffilyfe penso que retornar **404** seja somente no caso de não existir o recurso e não é o caso o recurso existe, porém estamos com erro ao acessa-ló! 

Concordo com o @gustavofonseca, devemos retorna erro com código **500**. 

Vejam como a mozilla defini esses erros: 

- https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Status/500
- https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Status/404

Reply:
> @gustavofonseca @joffilyfe penso que retornar **404** seja somente no caso de não existir o recurso e não é o caso o recurso existe, porém estamos com erro ao acessa-ló!
> 
> Concordo com o @gustavofonseca, devemos retorna erro com código **500**.
> 
> Vejam como a mozilla defini esses códigos:
> 
>     * https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Status/500
> 
>     * https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Status/404
> 
> 
> Mas minha justificativa vai de encontro a interface do site para com o Web sendo consistente nas repostas, precisamos verificar se iremos considerar as particularidade apontadas na resposta do @joffilyfe!, entendo que isso seja partidularidades da aplicação que talvez tenhamos que avaliar melhorar para termos retornos adequados.

@jamilatta a view retorna `404` apenas quando o recurso não existe no ponto de integração, é isso o que o código está fazendo depois da alteração deste pull request. Quanto ao código `500`, o pull requets passa a retornar em caso de falha entre o website e os pontos de integração, assim como pedido pelo issue #1649 e pelo bom senso na criação de sistemas web.

Reply:
Sim!  

O problema é que está vazando uma característica interna do sistema para os retornos **HTTP**.

Talvez tenhamos que alterar o comportamento da página do artigo para que ela seja exibida mesmo que não consiga obter qualquer recurso no ponto de integração, retornando assim um **200**, pois o recurso existe!

Dessa forma podemos colocar uma informação de que tivemos essa falha de integração na página do artigo, mas considero isso uma outra atividade, se assim for decidido.

Reply:
@jamilatta o recurso pode não existir porque o site está em um estado eventualmente inconsistente, é o caso do segundo fluxo que eu citei mais acima. Não daria para retornar `200` porque um dos recursos que podem não estar disponível é o próprio XML. Concordo com o vazamento de características inerentes de integração dos sistemas mas nós já estamos vazando isso com o código `500`.

Reply:
Qual ponto você acredita que inviabiliza retornarmos 200, detalhe não necessariamente precisamos exibir o artigo em caso de falha, podemos colocar uma mensagem de que não foi possível exibir o artigo! mas ficamos em conformidade com os retornos.

Não sei também se os indexadores irão revisitar páginas que retornaram 404! Realmente não sei! 

Lembrando que precisamos atender não somente usuário(pessoas) como usuário(máquinas).

Reply:
> @gustavofonseca do meu ponto de vista é sim necessário retornar `404` para alguns casos, estes referenciados pelo erro `NonRetryableError`. Temos o seguinte caminho alternativo:
> 
>     1. O Airflow registrou o documento na base do website;
> 
>     2. O documento passa a ficar disponível na interface pública;
> 
>     3. Por alguma falha humana ou não, o XML foi removido do _object storage_;
> 
>     4. O documento continua disponível no Website mas não mais disponível via integração;
> 
>     5. É necessário diferenciar erros intermitentes de erros relacionados a não existência do recurso;
> 
A não existência do recurso no _object store_ não é exatamente um erro, mas um dos resultados esperados. Considerei o caso descrito como uma inconsistência, daí o erro 500. Mas entendo a abordagem que vc propõe e penso que podemos seguir com ela.

> 
> Um outro ponto fluxo alternativo:
> 
>     1. O Airflow registrou o documento na base do website;
> 
>     2. O documento passa a ficar disponível na interface pública;
> 
>     3. O documento foi marcado como deletado no Kernel;
> 
>     4. Até que a task de sincronização seja executada o documento continua disponível no website;
> 
>     5. Ao acessar o documento via website nós teremos um `410` retornando do Kernel;
> 
>     6. É necessário diferenciar entre um erro casual como _timeout_ e remoção de recursos do ponto de integração;
> 
> 
> A questão é que o retorno do minio/kernel não são falhas e sim estados dos objetos armazenados nestes pontos. Se nós passamos a retornar apenas `500` nós criamos uma maquiagem nas entre o website e os pontos de integração, quando e se precisarmos debugar nós passamos a ter um caminho mais longo.

De fato, concordo que é importante diferenciar documentos deletados (que não é um erro) das demais condições generalizadas como `NonRetryableError`, mas podemos detalhar melhor em outro ticket.